### PR TITLE
Validations and Testing

### PR DESCRIPTION
**Pull request by Solobarine**
This pull request is to review the following done on the project:
- Validations were written for the models.
- Spec tests were also added.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+
+  validates :name, presence: true
+  validates :posts_counter, comparison: { greater_than_or_equal_to: 0 }
+
```
Comment (Kamba56) [inline app/models/user.rb]: - [ ]  Kindly add validation to validate if post_counter is an integer or not. GOod job adding validations for it to be greater than 0. You can use
```ruby
validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  validates :title, presence: true
+  validates :title, length: { maximum: 250 }
+  validates :comments_counter, comparison: { greater_than_or_equal_to: 0 }
+  validates :likes_counter, comparison: { greater_than_or_equal_to: 0 }
```
Comment (Kamba56) [inline app/models/post.rb]: - [ ] Kindly validate comments_counter and likes_counter to be of type integer too. This is demanded by the project requirements.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Solobarine, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job adding validations to your model data!👏👏👏
- [x] GOod job adding rspec tests to test validations and custom model methods

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Favourezeugwa — APPROVED** [review]
## STATUS: APPROVED 🥇 🏆 🍹 

Hi @Solobarine ,

I am Favour 👩🏽 , your reviewer for this project milestone.

Great Job implementing the required changes suggested the by the prevopis reviewer. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights 💯 
- No Linter errors ✔️
- Added validations to model data ✔️
- RSpec used to validations ✔️
- Descriptive PR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**